### PR TITLE
Add ntuplization code for sagitta-bias studies to `Alignment/OfflineValidation`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The goal of this PR is to add to the `Alignment/OfflineValidation` package a simple ntuplization code to do "sagitta bias" weak mode studies at CMS using di-muon resonant decays data as the ones in this [set of slides](https://indico.cern.ch/event/1271470/contributions/5380304/attachments/2639111/4566691/mm_TkAlMeeting_May2_2023.pdf). 
The tool is not interlaced yet in the all-in-one meta-data tool, but might be included at a later time.
I have added the new analyzer to the standard battery of unit tests.

#### PR validation:

Run extensive private tests, also run successfully:
   * `scram b runtests_Miscellanea`
   * `scram b runtests_testTrackAnalysis`
   
#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but will be backported to CMSSW_14_0_X for 2024 data-taking operations

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+    theZMuonVector.emplace_back(myGoodMuonVector[0]);
+
+    // do the matching of Z muons with inner tracks
+    unsigned int i = 0;
+    for (const auto& muon : theZMuonVector) {
+      i++;
+      float minD = 1000.;
+      const reco::Track* theMatch = nullptr;
+      for (const auto& track : event.get(tracksToken_)) {
+        float D = ::deltaR(muon->eta(), muon->phi(), track.eta(), track.phi());
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc:312]: `deltaR2` is preferable in terms of performance

File: Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+        if (g->status() != 1)
+          continue;
+
+        if (std::abs(g->pdgId()) != 13)
+          continue;
+
+        if (g->charge() != track->charge())
+          continue;
+
+        float dR = reco::deltaR(*g, *track);
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc:464]: same here

File: Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+  math::XYZVector vec2(trk2->px(), trk2->py(), trk2->pz());
+  return vec1.Dot(vec2) / (vec1.R() * vec2.R());
+}
+
+/*
+double SagittaBiasNtuplizer::openingAngle(const reco::Track& track1, const reco::Track& track2) {
+  double dPhi = std::abs(track1.phi() - track2.phi());
+  double dEta = std::abs(track1.eta() - track2.eta());
+  double deltaR = reco::deltaR(track1, track2);
+  double openingAngle = 2 * std::atan2(std::sqrt(std::sin(dEta / 2) * std::sin(dEta / 2) + std::sinh(dPhi / 2) * std::sinh(dPhi / 2)), std::sqrt(1 - deltaR * deltaR));
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc:517]: It is commented out, but...
```suggestion
  double deltaR2 = reco::deltaR2(track1, track2);
  double openingAngle = 2 * std::atan2(std::sqrt(std::sin(dEta / 2) * std::sin(dEta / 2) + std::sinh(dPhi / 2) * std::sinh(dPhi / 2)), std::sqrt(1 - deltaR2));
```

File: Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc (hunk 1 above)
Comment (mmusich) [reply Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc]: OK, I don't think performance here is of paramount importance, but in any case it was applied in the last push.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+1

**Pull request merged**